terraform/sponsored_bucket.tf

```diff
@@ -158,3 +159,39 @@ data "aws_iam_policy_document" "sponsored_bucket" {
     ]
   }
 }
+
+module "sponsored_embargo_bucket" {
+  source              = "./modules/dandiset_bucket"
+  bucket_name         = "dandiarchive-embargo"
+  versioning          = false
+  log_bucket_name     = "dandiarchive-embargo-logs"
+  log_bucket_owner_id = data.aws_canonical_user_id.sponsored_account.id
+  policy_json         = data.aws_iam_policy_document.sponsored_embargo_bucket.json
+  providers = {
+    aws = aws.sponsored
+  }
+}
+
+data "aws_iam_policy_document" "sponsored_embargo_bucket" {
+  version = "2008-10-17"
+
+  statement {
+    sid = "dandi-api"
+    principals {
```

Comment:
Note, you could also apply this policy to the IAM User itself. Technically, it doesn't matter, but it might provide better abstraction if the bucket doesn't need to know about the API User.